src/app/static/src/app/components/basic/Basic.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+    <div>
+        <h1>{{title}}</h1>
+        <div>App Type: {{appType}}</div>
+        <div>Basic Prop: {{basicProp}}</div>
+    </div>
+</template>
```

Comment:
So presumably each of these vue components will be different enough to justify them all being separate and not just configuring one component?

Reply:
Well, that's the idea. It may be that they eventually end up being so similar that keeping them in difference js libraries doesn't make sense, and maybe they even end up in the same component (just with slightly different configuration provided through the state). They will certainly share some child components.